### ISSUE TITLE
Loading xy.ltxml instead of xy.tex

### ISSUE THREAD

**Issue by jbs1** [jbs1/tract2#28]
_migrated from Trac, where originally posted by **ionita** on 29-Jan-2007 6:21pm_

We can follow the example of amstex, for which a .latexml file has been created which redirects the loading to the corresponding .ltxml.

Now we have to implement \xyoption which always follows \input xy in the respective papers.

**Comment by jbs1** [comment]
_migrated from Trac, where originally posted by **ionita** on 29-Jan-2007 6:21pm_

Some papers contain
\input xy

but in xy.tex (the xy-pic texinputs has been recently added to trunk/sty) the first (uncommented) line is:
\ifx\xyloaded\undefined\else\message{not reloaded}\endinput\fi

and the loading prints "not reloaded" in the stderr.log file, with any further xy related macros being unrecognized.

We probably want to redirect it to a xy.ltxml file.

**Comment by jbs1** [comment]
_migrated from Trac, where originally posted by **ionita** on 26-Feb-2007 12:34pm_

The creation of .latexml file for a package that has .tex description is also mentioned in LaTeXML doc: http://dlmf.nist.gov/LaTeXML/perldoc/LaTeXML/Package.html#description

**Comment by jbs1** [comment]
_migrated from Trac, where originally posted by **deyan** on 28-Jun-2010 2:14pm_

(In #1400) I will use this as the primary xy ticket (blocked by #1370 and #1318)

**Comment by jbs1** [comment]
_migrated from Trac, where originally posted by **deyan** on 28-Jun-2010 2:28pm_

To clarify the points raised here:
 * If one desires to make a binding for `foo.tex`, the created file should be `foo.tex.ltxml`. Any dependency that requires `foo.tex` can be realized via
```
RequirePackage('foo',type=>'tex');
```
 * LaTeXML can natively handle \ifx, \fi and friends, so that is not a problem.
 * The \xyloaded not being set is a problem, but easy to fix. We just need to define \xyloaded within the top XY binding via a DefMacro or so.
 * The input problem is more interesting. In principle there is no way to distinguish whether a user is inputing some source component or actually requiring a style file. My best suggestion is to always check when including `foo.tex` whether a `foo.tex.ltxml` exists, in which case we load the binding, rather than including the tex file.

I will set this up.

**Comment by jbs1** [comment]
_migrated from Trac, where originally posted by **deyan** on 31-Jan-2013 4:49pm_

The new effort to load xy.sty and so forth natively renders this ticket obsolete. Closing.
